admin/app/controllers/cache/PageDecacheController.scala

```diff
@@ -22,50 +22,20 @@ class PageDecacheController(wsClient: WSClient)(implicit context: ApplicationCon
 
   val authActions = new AuthActions(wsClient)
 
-  def renderPageDecache(url: Option[String] = None) = Action.async { implicit request =>
-    url match {
-      case Some(s) => renderPrePurgeTestResult(s)
-      case None => Future(NoCache(Ok(views.html.cache.pageDecache())))
-    }
-  }
-
-  private def renderPrePurgeTestResult(purgeUrl: String)(implicit request: Request[AnyContent]) = {
-    getRouterUrl(purgeUrl).map { routerUrl =>
-      wsClient.url(routerUrl)
-        .withRequestTimeout(2.second)
-        .withVirtualHost("www.theguardian.com")
-        .withFollowRedirects(true)
-        .get()
-        .map { response =>
-          PrePurgeTestResult(purgeUrl, response.status >= 200 && response.status < 300)
-        }.recoverWith {
-          case t: Throwable => successful(PrePurgeTestResult(purgeUrl, passed = false))
-        }.map { result =>
-          NoCache(Ok(views.html.cache.pageDecache(Some(result))))
-        }
-    }.getOrElse(successful(InternalServerError("Couldn't get router URL - please go back and try again")))
+  def renderPageDecache() = Action.async { implicit request =>
+      Future(NoCache(Ok(views.html.cache.pageDecache())))
   }
 
   def decache() = authActions.AuthActionTest.async { implicit request =>
     getSubmittedUrl(request).map(new URI(_)).map{ urlToDecache =>
-
       new CdnPurge(wsClient)
         .soft(SurrogateKey(urlToDecache.getPath))
-        .map { purgeSent =>
-          val message = if(purgeSent) "Purge request successfully sent" else "Purge request was not successful, please report this issue"
-          NoCache(Ok(views.html.cache.pageDecache(None, message)))
-        }
+        .map { _ => "Purge request successfully sent" }
+        .recover { case e => s"Purge request was not successful, please report this issue: '${e.getLocalizedMessage}'" }
+        .map { message => NoCache(Ok(views.html.cache.pageDecache(message))) }
```

Comment:
Fist `map` and `recover` transform the result (success and failure) into a message. 2nd `map` is transforming the message into some html response.
If we were on Scala 2.12 I would be using [transform](http://www.scala-lang.org/api/2.12.1/scala/concurrent/Future.html#transformWith[S](f:scala.util.Try[T]=>scala.concurrent.Future[S])(implicitexecutor:scala.concurrent.ExecutionContext):scala.concurrent.Future[S])